CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/clean/MergeTransactionsNGTest.java

```diff
@@ -584,4 +597,128 @@ public void testMergeTransactionSomeSelected() throws Exception {
 
         assertEquals(1, mergedCount);
     }
+
+    @Test
+    public void testSortTransactionsWithNullType1() {
+        MergeTransactionType instance = new MergeTransactionType() {
+            @Override
+            public String getName() {
+                throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
+            }
+
+            @Override
+            public void updateParameters(Map<String, PluginParameter<?>> parameters) {
+                throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
+            }
+
+            @Override
+            public Map<Integer, Set<Integer>> getTransactionsToMerge(GraphWriteMethods graph, Comparator<Long> leadTransactionChooser, int threshold, boolean selectedOnly) throws MergeTransactionType.MergeException {
+                throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
+            }
+        };
+
+        Integer[] transactions = new Integer[2];
+        transactions[0] = txId4;
+        transactions[1] = txId2;
+
+        Integer[] expecting = new Integer[2];
+        expecting[0] = txId4;
+        expecting[1] = txId2;
+
+        instance.sortTransactions(transactions, graph, transactionTypeAttribute, transactionDateTimeAttribute, MergeTransactionsPlugin.LATEST_TRANSACTION_CHOOSER);
+        assertArrayEquals(expecting, transactions);
+    }
+
+    @Test
+    public void testSortTransactionsWithNullType2() {
+        MergeTransactionType instance = new MergeTransactionType() {
```

Comment:
It looks like there a few instances of creating a new MergeTransactionType and having to implement all the necessary functions (all implemented the same way). Instead of doing this, I would create a private implementation (like TestMergeTransactionType) and then your unit tests can use that one.